src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNIMarsHandle.cs

```diff
@@ -81,9 +82,11 @@ public SNIMarsHandle(SNIMarsConnection connection, ushort sessionId, object call
         {
             _sessionId = sessionId;
             _connection = connection;
+            _connectionId = connection.ConnectionId;
```

Comment:
I'm not sure about this one. Because there are multiple mars handles per physical handle this will put data in the traces for all of them with the same if. It might be better to keep the generated uniqueid for the children and also print the parent connection id in the trace messages.

Reply:
Yes, but the parent connection Id does not change and the `sessionId` is what differentiates various handles here I think. I will revisit all Ids being logged as well, to double check but it certainly does help in tracking Connection's activity.